core/src/main/java/org/elasticsearch/search/internal/DefaultSearchContext.java

```diff
@@ -179,12 +193,20 @@ public void doClose() {
      */
     @Override
     public void preProcess() {
-        if (!(from() == -1 && size() == -1)) {
-            // from and size have been set.
-            int numHits = from() + size();
-            if (numHits < 0) {
-                String msg = "Result window is too large, from + size must be less than or equal to: [" + Integer.MAX_VALUE + "] but was [" + (((long) from()) + ((long) size())) + "]";
-                throw new QueryPhaseExecutionException(this, msg);
+        if (scrollContext == null) {
+            long from = from() == -1 ? 0 : from();
+            long size = size() == -1 ? 10 : size();
+            long resultWindow = from + size;
+            // We need settingsService's view of the settings because its dynamic.
+            // indexService's isn't.
+            int maxResultWindow = indexService.settingsService().getSettings().getAsInt(MAX_RESULT_WINDOW, Defaults.MAX_RESULT_WINDOW);
```

Comment:
So I know we don't expect anyone to set this to a crazy high value, but I think we should make it a `long` **just** because someone is liable to set it to `Integer.MAX_VALUE + 1` and then wonder why it shows negative due to wrapping.